protos/file.proto

```diff
@@ -173,4 +173,17 @@ message Field {
 
   // optional field metadata (e.g. extension type name/parameters)
   map<string, bytes> metadata = 10;
+
+  /// The storage class of the field
+  ///
+  /// This determines the rate at which the field is compacted.
+  ///
+  /// Currently, there are only two storage classes:
+  /// 
+  /// "" - The default storage class.
+  /// "blob" - The field is compacted into fewer rows per fragment.
```

Comment:
is the criteria for this entirely based on size of row values? or do you imagine there will eventually be other considerations?

Reply:
Size is the only thing I can think of and I don't expect us to need more than two storage classes for the next year or two.

It's possible we have a special "foreign" storage class or something like that for values stored in remote datasets but I think that would be best solved in a different way.

At the moment the only other class I can really think of is for extremely large values where file-per-value makes more sense.

I also hadn't quite done the math yet.  We want to balance:
* We can fit 8MiB worth of values into a file (min bound on num_values)
* We don't have more than 10GiB of values in a file (max bound on num_values)

For example, if we start with int32 we need at least 2Mi values (ideally more actually due to compression).  Given 10GiB this gives us a surprisingly small max size of 5KiB (anything larger will occupy more than 10GiB for 2Mi values).

If we are generous and make the cutoff at 64KiB then we need to make sure we have at least 128 values per file which should allow us up to 160MIB per value.

If we find that we can't be generous and we really do want the cutoff at...say...8KiB then we need at least 1024 values per file which means no more than 10MiB per value which is starting to get to the point where we might want three storage classes.

Reply:
> If we find that we can't be generous and we really do want the cutoff at...say...8KiB

This logic is based on the 10GiB limit which is a somewhat arbitrary "let's pretend we have at most 100 or so 'big' columns and we want to keep files under 1TiB" criteria which seems a little conservative.  At 64KiB we're fine as long as we only have 1 or 2 big columns or can tolerate files over 1TiB.